dcp/api/job.py

```diff
@@ -167,14 +167,28 @@ def wait(self):
             return dry.aio.blockify(self._wait)()
 
         def on(self, *args):
+            # deserialize job on event parameters before passing them to user defined callback
+            def cb_deserialize_wrapper(callback):
+                def new_cb(*inner_args):
+                    nonlocal callback
```

Comment:
Do we need to change `callback`? If not, it's probably better to remove this line.
```suggestion
```